File: algorithm/src/main/java/baekjoon/p01964/Main.java
package baekjoon.p01964;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.nextLong()));
    }

    private static long solve(long n) {
        long sum = 5;
        for (long i = 2; i <= n; i++) {
            sum += (i * 2 + i + 1);
        }

        return sum % 45678;
    }

}

File: algorithm/src/main/java/baekjoon/p01964/Main.kt
package baekjoon.p01964

fun main(args: Array<String>) {
    println(solve(readLine()!!.toLong()))
}

private fun solve(n: Long): Long {
    var sum = 5L
    for (i in 2..n) {
        sum += (i * 2 + i + 1)
    }

    return sum % 45678
}